src/SFA.DAS.EmployerAccounts.Events/Messages/AgreementSignedMessage.cs

```diff
@@ -1,12 +1,26 @@
-﻿using SFA.DAS.Messaging.Attributes;
+﻿using System;
+using SFA.DAS.Messaging.Attributes;
 
 namespace SFA.DAS.EmployerAccounts.Events.Messages
 {
     [MessageGroup("agreement_signed")]
-    public class AgreementSignedMessage
+    public class AgreementSignedMessage : Message
     {
-        public long AccountId { get; set; }
-        public long LegalEntityId { get; set; }
-        public long AgreementId { get; set; }
+        public AgreementSignedMessage()
+        {
+
+        }
+
+        public AgreementSignedMessage(long accountId, long aggreementId, string providerName, string signedByName, long legalEntityId) :base(signedByName, accountId)
+        {
+            AgreementId = aggreementId;
+            ProviderName = providerName;
+            LegalEntityId = legalEntityId;
+        }
+
+        public string ProviderName { get; }
```

Comment:
For consistency i'd move the public params above the constructors as this is how its been done elsewhere. i'll only mention it here rather than on each message

Reply:
ok